From 40a431be1222ae015ef1338abd65c292a45fe50b Mon Sep 17 00:00:00 2001
From: Noriko Hosoi <nhosoi@redhat.com>
Date: Wed, 29 Jul 2015 11:26:22 -0700
Subject: [PATCH 353/363] Ticket #48232 - winsync lastlogon attribute not
 syncing between DS and AD.

Bug Description:
From Microsoft forum:
 The DirSync control taps into the replication stream to get the necessary
 changes. Since lastLogon is not replicated, it isn't available via the
 DirSync control.
Additional notes:
 The lastLogon attribute is not replicated.
 In contrast the lastLogontimeStamp attribute is replicated.

Fix Description:
 Instead of lastLogon|lastLogoff, sync lastLogonTimestamp|lastLogoffTimestamp
 which are the target of DirSync, to ntUserLastLogon|ntUserLastLogoff.

https://fedorahosted.org/389/ticket/48232

Reviewed by rmeggins@redhat.com (Thank you, Rich!!)

(cherry picked from commit 0db29788e6c1b17f944fcafa368b66580e1e90d5)
(cherry picked from commit b81adb0bc8ad97fec50fba30454e94858476bad5)
(cherry picked from commit a3af2351bb1fb3782c9f79209043ddc36e0e06d1)
(cherry picked from commit e94bf22c252197c85e7f7562d3ba93f5a0f2215e)
---
 ldap/servers/plugins/replication/windows_protocol_util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ldap/servers/plugins/replication/windows_protocol_util.c b/ldap/servers/plugins/replication/windows_protocol_util.c
index 12acf94..31c9495 100644
--- a/ldap/servers/plugins/replication/windows_protocol_util.c
+++ b/ldap/servers/plugins/replication/windows_protocol_util.c
@@ -223,8 +223,8 @@ static windows_attribute_map user_attribute_map[] =
 {
 	{ "homeDirectory", "ntUserHomeDir", bidirectional, always, normal},
 	{ "scriptPath", "ntUserScriptPath", bidirectional, always, normal},
-	{ "lastLogon", "ntUserLastLogon", fromwindowsonly, always, normal},
-	{ "lastLogoff", "ntUserLastLogoff", fromwindowsonly, always, normal},
+	{ "lastLogonTimestamp", "ntUserLastLogon", fromwindowsonly, always, normal},
+	{ "lastLogoffTimestamp", "ntUserLastLogoff", fromwindowsonly, always, normal},
 	{ "accountExpires", "ntUserAcctExpires", bidirectional, always, normal},
 	{ "codePage", "ntUserCodePage", bidirectional, always, normal},
 	{ "logonHours", "ntUserLogonHours", bidirectional, always, normal},
-- 
2.4.3

